File: expenses-manager-backend-java/src/main/java/com/github/beliakou/expenses/controller/UserController.java
package com.github.beliakou.expenses.controller;

import com.github.beliakou.expenses.domain.User;
import com.github.beliakou.expenses.repository.UserRepository;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class UserController {

    private UserRepository userRepository;

    public UserController(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @GetMapping("/user")
    public User getUser(@RequestParam("name") String username) {
        return this.userRepository.findUser(username).orElseThrow(RuntimeException::new);
    }
}

File: expenses-manager-backend/expenses-manager-server/src/main/kotlin/com/github/beliakou/expenses/server/controller/UserController.kt
package com.github.beliakou.expenses.server.controller

import com.github.beliakou.expenses.server.model.User
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.mongodb.core.MongoTemplate
import org.springframework.data.mongodb.core.query.Criteria
import org.springframework.data.mongodb.core.query.Query
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
class UserController(@Autowired val mongoTemplate: MongoTemplate) {

    @GetMapping("/user")
    fun user(@RequestParam(value = "name") name: String): User? {
        return this.mongoTemplate.findOne(
                Query.query(Criteria.where("name").`is`(name)), User::class.java, "user")
    }
}